skiko/src/awtMain/objectiveC/macos/MetalRedrawer.mm

```diff
@@ -216,14 +216,16 @@ JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_redrawer_MetalRedrawer_createMe
         device.layer.pixelFormat = MTLPixelFormatBGRA8Unorm;
         device.layer.contentsGravity = kCAGravityTopLeft;
 
-        CGFloat transparent[] = { 0.0f, 0.0f, 0.0f, 0.0f };
-        device.layer.backgroundColor = CGColorCreate(CGColorSpaceCreateDeviceRGB(), transparent);
-        device.layer.opaque = NO;
+        NSWindow* window = (__bridge NSWindow*) (void *) windowPtr;
 
+        device.layer.opaque = NO;
         if (transparency)
         {
-            NSWindow* window = (__bridge NSWindow*) (void *) windowPtr;
+            CGFloat transparent[] = { 0.0f, 0.0f, 0.0f, 0.0f };
+            device.layer.backgroundColor = CGColorCreate(CGColorSpaceCreateDeviceRGB(), transparent);
             window.hasShadow = NO;
+        } else {
+            device.layer.backgroundColor = window.backgroundColor.CGColor;
```

Comment:
Compose uses SkiaLayer not only in the whole window, but also in individual parts (like `ComposePanel` that can be embedded into IDEA).

This will not allow to use transparent `ComposePanel`'s.

The fact that MetalRedrawer was coupled with `window` for transparency was unfortunate, and we should fix that in the future.

Can we use `SkiaLayer`'s color instead?

Reply:
Yes, we probably can, but this change would be not so trivial. So I would prefer to postpone it until that decoupling refactoring.

Reply:
> until that decoupling refactoring.

It wasn't planned, and we shouldn't wait for it to implement this feature. In anyway, we will need to read a background of SkiaLayer, and pass it to Redrawer, and then to the native code

Reply:
The PR will look like [this](https://github.com/JetBrains/skiko/pull/677/files). Except, we will read and set background